### PR TITLE
Install latest PennyLane for CI

### PR DESCRIPTION
**Pull request by astralcai**

### CODE REVIEW

**Comment by astralcai** [comment]
[sc-63171]

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -50,7 +51,7 @@ jobs:
 
     strategy:
       matrix:
-        python-version: ["3.9", "3.11"]
+        python-version: ["3.9", "3.10", "3.11"]
```
Comment (dwierichs) [inline .github/workflows/tests.yml:54]: Why was 3.10 not included here, previously? :thinking:

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -75,6 +76,7 @@ jobs:
         run: |
           python -m pip install --upgrade pip
           pip install '.[test]'
+          pip install --upgrade git+https://github.com/PennyLaneAI/pennylane.git#egg=pennylane
```
Comment (dwierichs) [inline .github/workflows/tests.yml:79]: Should we keep the ordering identical between tests and integration-tests :thinking:

**Review by dwierichs — APPROVED** [review]
LGTM, just two tiny questions mostly out of curiousity :)

File: .github/workflows/tests.yml (hunk 1 above)
Comment (astralcai) [reply .github/workflows/tests.yml:54]: So it was previously "3.8" and "3.10", and then it got changed to "3.9" and "3.11". I don't see why "3.10" shouldn't be included so I added it back in.

File: .github/workflows/tests.yml (hunk 2 above)
Comment (astralcai) [inline .github/workflows/tests.yml:79]:
```suggestion
          pip install --upgrade git+https://github.com/PennyLaneAI/pennylane.git#egg=pennylane
          pip install '.[test]'
```

**Pull request merged**